### DISCUSSION TITLE
Change volume, brightness, Kelvin, and anything else that uses a 0-1 range to use a 0-100 range across all integrations/platforms

### DISCUSSION THREAD

**Discussion by raman325** [home-assistant/architecture#587] [Entity Models]
Through conversations with a wide number of contributors, I have observed a general consensus that 0-100% makes more intuitive sense than the numeric multiplier that we currently use (0-1) for things which have a relative value. 

Considerations:
- It appears that the majority of integrations/platforms that have these attributes use 0-1 today although there are a couple of integrations that use 0-100 for custom services.
- When a sensor has a unit of measurement as `%`, the actual sensor state is the whole number representation of that percentage, e.g. 90 for `90%`. So this is already confusing, and standardizing on a single strategy (0-100) will simplify things for users. Case in point, I only learned about this behavior with sensors because I assumed % sensors required 0-1 and wrote an integration with that assumption in mind, which turned out to be incorrect.

Impact Area:
- Look for any places that `cv.small_float` is used in schema validation for potential target areas
- This will introduce breaking changes for existing automations
- Frontend will likely require coordinated changes with backend to ensure smooth transition
- When changing platform behavior, all consuming integrations will have to be reviewed and possibly updated to avoid breaking the integration

I am not volunteering to do this work at the moment, just putting it out there for the community to get buy in and possibly (hopefully?) recruit volunteers who would be willing to take this on

**Comment by amelchio** [comment]
To clarify, brightness and color temperature are not currently on a 0-1 range. However, when the confusing volume scale was brought up in Discord, I mentioned the two light attributes as other examples of weird units. I think they should be migrated to percentage and [Kelvin](https://github.com/home-assistant/architecture/discussions/564) respectively.
